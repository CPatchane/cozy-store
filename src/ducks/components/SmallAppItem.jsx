import React from 'react'

import Icon from 'cozy-ui/react/Icon'
import { translate } from 'cozy-ui/react/I18n'

import defaultAppIcon from '../../assets/icons/icon-cube.svg'
import { Link } from 'react-router-dom'
import Button from 'cozy-ui/react/Button'

export const SmallAppItem = ({
  t,
  slug,
  developer,
  editor,
  icon,
  name,
  namePrefix,
  installed,
  onClick,
  installedAppLink
}) => {
  const onShortcutAppButton = (e, link) => {
    // prevent from opening parent SmallAppItem link
    // which will lead to the application main page
    e.stopPropagation()
    if (link) window.location.assign(link)
  }
  return (
    // HACK a11y
    // `onKeyDown={(e) => e.keyCode === 13 ? onClick() : null`
    // This syntax make the div accessible.
    <div
      className='sto-small-app-item'
      onKeyDown={e => (e.keyCode === 13 ? onClick() : null)}
      onClick={onClick}
      tabIndex={0}
    >
      {icon ? (
        <img
          src={icon}
          alt={`${slug}-icon`}
          width='64'
          height='64'
          className='sto-small-app-item-icon'
        />
      ) : (
        <Icon
          className='sto-small-app-item-icon--default blurry'
          width='48'
          height='64'
          icon={defaultAppIcon}
          color='#95999D'
        />
      )}
      {installed && (
        <Icon
          className='sto-small-app-item-badge-installed'
          icon='check'
          color='#2bba40'
        />
      )}
      <div className='sto-small-app-item-desc'>
        <h4 className='sto-small-app-item-title'>
          {namePrefix && `${namePrefix} `}{name}
        </h4>
        <p className='sto-small-app-item-detail'>
          {developer.name === 'Cozy' ? 'Cozy Cloud Inc.' : developer.name}
        </p>
        {installed
          ? <Button
            onClick={(e) => onShortcutAppButton(e, installedAppLink)}
            label={t('app_item.open')}
            theme='secondary'
            className='sto-small-app-item-button'
            size='tiny'
          />
          : <Link
            to={`/discover/${slug}/manage`}
            onClick={(e) => onShortcutAppButton(e)}
            className='c-btn c-btn--regular c-btn--tiny sto-small-app-item-button sto-small-app-item-button-install'
          >
            <span>{t('app_item.install')}</span>
          </Link>
        }
      </div>
    </div>
  )
}

export default translate()(SmallAppItem)
