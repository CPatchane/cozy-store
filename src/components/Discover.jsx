import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'

import SmallAppItem from './SmallAppItem'

const DEMO_APPS = [
  {
    slug: 'bank',
    name: 'Bank',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-bank.svg')
  },
  {
    slug: 'sante',
    name: 'Sant√©',
    developer: {name: 'MAIF Ltd'},
    editor: '',
    icon: require('../assets/temp_apps/icon-sante.svg')
  },
  {
    slug: 'mon_logis',
    name: 'Mon Logis',
    developer: {name: 'HoodBrains'},
    editor: '',
    icon: require('../assets/temp_apps/icon-mon_logis.svg')
  },
  {
    slug: 'calendar',
    name: 'Calendar',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-calendar.svg')
  }
]

class Discover extends Component {
  render () {
    const { t } = this.props
    return (
      <div className='sto-discover sto-soon'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {DEMO_APPS.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
              />
            })}
          </div>
        </div>
      </div>
    )
  }
}

export default translate()(Discover)
