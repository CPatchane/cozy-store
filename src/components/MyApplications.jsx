import React, { Component } from 'react'

import { Route } from 'react-router-dom'
import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from './SmallAppItem'
import ApplicationModal from './ApplicationModal'

class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}/manage`)
  }

  render () {
    const { t, myApps, isFetching, error } = this.props
    return (
      <div className='sto-myapps'>
        <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
        <div className='sto-myapps-list'>
          {!isFetching && myApps && !!myApps.length &&
            myApps.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
                onClick={() => this.onAppClick(a.slug)}
              />
            })
          }
          {error &&
            <p>{error}</p>
          }
        </div>
        {isFetching &&
          <Spinner
            size='xxlarge'
            loadingType='appsFetching'
            middle='true'
          />
        }

        <Route path='/myapps/:appSlug/manage' render={({ match }) =>
          (<ApplicationModal {...this.props} match={match} />)
        } />
      </div>
    )
  }
}

export default translate()(MyApplications)
