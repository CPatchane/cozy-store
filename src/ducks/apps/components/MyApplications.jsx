import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import ApplicationRouting from './ApplicationRouting'
import SmallAppItem from '../../components/SmallAppItem'

import { getLocalizedAppProperty } from 'ducks/apps'

export class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchInstalledApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}`)
  }

  render () {
    const { t, lang, installedApps, isFetching, fetchError } = this.props
    return (
      <div className='sto-myapps'>
        {this.props.match.isExact ? (
          <div>
            <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
            <div className='sto-myapps-list'>
              {!isFetching &&
                installedApps &&
                !!installedApps.length &&
                installedApps.map(app => {
                  return (
                    <SmallAppItem
                      slug={app.slug}
                      developer={app.developer}
                      editor={app.editor}
                      icon={app.icon}
                      name={getLocalizedAppProperty(app, 'name', lang)}
                      version={app.version}
                      installed={app.installed}
                      onClick={() => this.onAppClick(app.slug)}
                      key={app.slug}
                    />
                  )
                })}
              {fetchError && <p className='coz-error'>{fetchError.message}</p>}
            </div>
          </div>
        ) : null}

        <ApplicationRouting
          installedApps={installedApps}
          isFetching={isFetching}
          installApp={this.props.installApp}
          uninstallApp={this.props.uninstallApp}
          parent='myapps'
        />

        {isFetching && (
          <Spinner size='xxlarge' loadingType='appsFetching' middle='true' />
        )}
      </div>
    )
  }
}

export default translate()(MyApplications)
