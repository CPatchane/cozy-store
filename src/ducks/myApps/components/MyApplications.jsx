import React, { Component } from 'react'

import { Route } from 'react-router-dom'
import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from '../../components/SmallAppItem'
import UninstallModal from './UninstallModal'

export class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchMyApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}/manage`)
  }

  render () {
    const { t, myApps, isFetching, error } = this.props
    return (
      <div className='sto-myapps'>
        <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
        <div className='sto-myapps-list'>
          {!isFetching && myApps && !!myApps.length &&
            myApps.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
                onClick={() => this.onAppClick(a.slug)}
              />
            })
          }
          {error &&
            <p>{error}</p>
          }
        </div>
        {isFetching &&
          <Spinner
            size='xxlarge'
            loadingType='appsFetching'
            middle='true'
          />
        }

        <Route path='/myapps/:appSlug/manage' render={({ match }) => {
          if (isFetching) return
          if (myApps.length) {
            return <UninstallModal {...this.props} match={match} />
          }
        }} />
      </div>
    )
  }
}

export default translate()(MyApplications)
