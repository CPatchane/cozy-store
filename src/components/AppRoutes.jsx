import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import MyApplications from '../containers/MyApplications'

const AppRoutes = () => {
  return <Switch>
    <Route path='/myapps' component={MyApplications} />
    <Redirect from='/' to='/myapps' />
  </Switch>
}

export default AppRoutes
