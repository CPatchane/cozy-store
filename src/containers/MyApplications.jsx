import { connect } from 'react-redux'

import { fetchApps, uninstallApp } from '../ducks/myApps'

import MyApplications from '../components/MyApplications'

const mapStateToProps = (state, ownProps) => ({
  myApps: state.myApps.list,
  isFetching: state.myApps.isFetching,
  error: state.myApps.error
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  fetchApps: () => dispatch(fetchApps()),
  uninstallApp: (appSlug) => dispatch(uninstallApp(appSlug))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MyApplications)
