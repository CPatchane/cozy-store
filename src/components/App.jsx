
import React from 'react'

import Sidebar from './Sidebar'
import AppRoutes from './AppRoutes'

const App = () => (
  <div className='sto-wrapper coz-sticky'>
    <Sidebar />
    <main className='sto-content'>
      <AppRoutes />
    </main>
  </div>
)
export default App
