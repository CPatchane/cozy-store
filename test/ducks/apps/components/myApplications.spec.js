'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { tMock } from '../../../jestLib/I18n'
import SmallAppItem from 'ducks/components/SmallAppItem'
import { MyApplications } from 'ducks/apps/components/MyApplications'

import mockApps from '../_mockApps'

Enzyme.configure({ adapter: new Adapter() })

const mockMyApplicationsError = new Error('This is a test error')

const mockInstalledApps = mockApps.filter(a => a.installed)

const getMockProps = (
  installedApps = mockInstalledApps,
  isFetching = false,
  fetchError = null,
  match = { isExact: true }
) => ({
  fetchInstalledApps: jest.fn(),
  uninstallApp: jest.fn(),
  installedApps,
  isFetching,
  fetchError,
  history: { push: jest.fn() },
  match
})

describe('MyApplications component', () => {
  it('should be rendered correctly if apps uninstalled', () => {
    const mockProps = getMockProps()
    const component = shallow(
      <MyApplications t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should render correctly a spinner if apps is fetching', () => {
    const mockProps = getMockProps([], true, null)
    const component = shallow(
      <MyApplications t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should display error from props correctly', () => {
    const mockProps = getMockProps([], false, mockMyApplicationsError)
    const component = shallow(
      <MyApplications t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should handle correctly items onClick', () => {
    const mockProps = getMockProps()
    const component = shallow(<MyApplications t={tMock} {...mockProps} />)
    expect(component.find(SmallAppItem).length).toBe(mockInstalledApps.length)
    const appItem = component
      .find(SmallAppItem)
      .at(0)
      .dive() // shallow on more level on first app item
    appItem.simulate('click')
    // history push to app modal URL
    expect(mockProps.history.push.mock.calls.length).toBe(1)
    expect(mockProps.history.push.mock.calls[0][0]).toBe(
      `/myapps/${mockInstalledApps[0].slug}`
    )
  })

  it('should render only routing if path not exactly match /myapps', () => {
    const mockProps = getMockProps(mockInstalledApps, false, null, {
      isExact: false
    })
    const component = shallow(
      <MyApplications t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})
