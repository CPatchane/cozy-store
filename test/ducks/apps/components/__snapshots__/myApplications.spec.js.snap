// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyApplications component should be rendered correctly if apps uninstalled 1`] = `
<div
  className="sto-myapps"
>
  <div>
    <div
      className="sto-myapps-sections"
    >
      <Wrapper
        apps={
          Array [
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Collect",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://collect.cozy.mock/",
              "slug": "collect",
              "tags": Array [
                "konnector",
                "collect",
                "bills",
                "providers",
                "files",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0",
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "name": "Drive",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://drive.cozy.mock/",
              "slug": "drive",
              "tags": Array [
                "search",
                "files",
                "folders",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0-beta89bnhj3993",
            },
            Object {
              "categories": Array [
                "transport",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Trinlane",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
              },
              "slug": "konnector-trinlane",
              "tags": Array [
                "transport",
                "files",
                "bills",
              ],
              "type": "konnector",
              "uninstallable": true,
              "version": "0.1.0",
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
          ]
        }
        error={null}
        onAppClick={[Function]}
      />
    </div>
  </div>
  <withRouter(ApplicationRouting)
    actionError={null}
    fetchLatestApp={undefined}
    installApp={undefined}
    installedApps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "name": "Drive",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://drive.cozy.mock/",
          "slug": "drive",
          "tags": Array [
            "search",
            "files",
            "folders",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0-beta89bnhj3993",
        },
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
      ]
    }
    isAppFetching={undefined}
    isFetching={false}
    parent="myapps"
    uninstallApp={[MockFunction]}
    updateApp={undefined}
  />
</div>
`;

exports[`MyApplications component should be rendered correctly with apps and URL search params 1`] = `
<div
  className="sto-myapps"
>
  <div>
    <div
      className="sto-myapps-sections"
    >
      <Wrapper
        apps={
          Array [
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Collect",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://collect.cozy.mock/",
              "slug": "collect",
              "tags": Array [
                "konnector",
                "collect",
                "bills",
                "providers",
                "files",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0",
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "name": "Drive",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://drive.cozy.mock/",
              "slug": "drive",
              "tags": Array [
                "search",
                "files",
                "folders",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0-beta89bnhj3993",
            },
          ]
        }
        error={null}
        onAppClick={[Function]}
      />
    </div>
  </div>
  <withRouter(ApplicationRouting)
    actionError={null}
    fetchLatestApp={undefined}
    installApp={undefined}
    installedApps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "name": "Drive",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://drive.cozy.mock/",
          "slug": "drive",
          "tags": Array [
            "search",
            "files",
            "folders",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0-beta89bnhj3993",
        },
      ]
    }
    isAppFetching={undefined}
    isFetching={false}
    parent="myapps"
    uninstallApp={[MockFunction]}
    updateApp={undefined}
  />
</div>
`;

exports[`MyApplications component should display error from props correctly 1`] = `
<div
  className="sto-myapps"
>
  <div>
    <div
      className="sto-myapps-sections"
    >
      <Wrapper
        apps={Array []}
        error={[Error: This is a test error]}
        onAppClick={[Function]}
      />
    </div>
  </div>
  <withRouter(ApplicationRouting)
    actionError={null}
    fetchLatestApp={undefined}
    installApp={undefined}
    installedApps={Array []}
    isAppFetching={undefined}
    isFetching={false}
    parent="myapps"
    uninstallApp={[MockFunction]}
    updateApp={undefined}
  />
</div>
`;

exports[`MyApplications component should render correctly a spinner if apps is fetching 1`] = `
<div
  className="sto-myapps"
>
  <div>
    <div
      className="sto-myapps-sections"
    />
  </div>
  <withRouter(ApplicationRouting)
    actionError={null}
    fetchLatestApp={undefined}
    installApp={undefined}
    installedApps={Array []}
    isAppFetching={undefined}
    isFetching={true}
    parent="myapps"
    uninstallApp={[MockFunction]}
    updateApp={undefined}
  />
  <Wrapper
    loadingType="appsFetching"
    middle={true}
    size="xxlarge"
  />
</div>
`;

exports[`MyApplications component should render only routing if path not exactly match /myapps 1`] = `
<div
  className="sto-myapps"
>
  <withRouter(ApplicationRouting)
    actionError={null}
    fetchLatestApp={undefined}
    installApp={undefined}
    installedApps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "name": "Drive",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://drive.cozy.mock/",
          "slug": "drive",
          "tags": Array [
            "search",
            "files",
            "folders",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0-beta89bnhj3993",
        },
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
      ]
    }
    isAppFetching={undefined}
    isFetching={false}
    parent="myapps"
    uninstallApp={[MockFunction]}
    updateApp={undefined}
  />
</div>
`;

exports[`MyApplications component should use BarCenter from cozy-bar in mobile view only 1`] = `
<div
  className="sto-myapps"
>
  <div>
    <BarCenter>
      <h2
        className="sto-view-title"
      >
        My Applications
      </h2>
    </BarCenter>
    <div
      className="sto-myapps-sections"
    >
      <Wrapper
        apps={
          Array [
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Collect",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://collect.cozy.mock/",
              "slug": "collect",
              "tags": Array [
                "konnector",
                "collect",
                "bills",
                "providers",
                "files",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0",
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "name": "Drive",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://drive.cozy.mock/",
              "slug": "drive",
              "tags": Array [
                "search",
                "files",
                "folders",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0-beta89bnhj3993",
            },
            Object {
              "categories": Array [
                "transport",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Trinlane",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
              },
              "slug": "konnector-trinlane",
              "tags": Array [
                "transport",
                "files",
                "bills",
              ],
              "type": "konnector",
              "uninstallable": true,
              "version": "0.1.0",
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
          ]
        }
        error={null}
        onAppClick={[Function]}
      />
    </div>
  </div>
  <withRouter(ApplicationRouting)
    actionError={null}
    fetchLatestApp={undefined}
    installApp={undefined}
    installedApps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "name": "Drive",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://drive.cozy.mock/",
          "slug": "drive",
          "tags": Array [
            "search",
            "files",
            "folders",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0-beta89bnhj3993",
        },
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
      ]
    }
    isAppFetching={undefined}
    isFetching={false}
    parent="myapps"
    uninstallApp={[MockFunction]}
    updateApp={undefined}
  />
</div>
`;
